src/containers/JoinSessionForm.js

```diff
@@ -53,26 +54,34 @@ const JoinSessionForm = (props) => {
   };
 
   const renderStageName = () => {
-    if (!sessionInfo) return <Redirect to="/session-not-found" />;
-    else return (
-      <div>
-        joining <span className="highlight">{sessionInfo.admin_name}</span>'s session.<br/>
-        please choose your username.<br/>
-        <br/>
-
-        <form onSubmit={handleSubmitName}>
-          <DefaultTextField
-            placeholder="username"
-            autofocus={true}
-            value={username}
-            onChange={updateField}
-            error={props.error}
-          />
+    // only redirect if we know the session isn't valid
+    // !sessionInfo doesn't suffice since the request to fetch the session info is asynchronous.
+    if (!sessionInfo && error.error === SESSION_NOT_FOUND_ERROR)
+      return <Redirect to="/session-not-found" />;
```

Comment:
don't we get `SESSION_NOT_FOUND_ERROR` from the same request we get `sessionInfo`

Reply:
Yeah, the `if (!sessionInfo)` is superfluous. I noticed it too, it was still there from the original PR.